staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -240,8 +246,11 @@ func (c *Controller) processNextWorkItem() bool {
 // syncHandler compares the actual state with the desired, and attempts to
 // converge the two. It then updates the Status block of the Foo resource
 // with the current status of the resource.
-func (c *Controller) syncHandler(key string) error {
+func (c *Controller) syncHandler(ctx context.Context, key string) error {
 	// Convert the namespace/name string into a distinct namespace and name
+	logger := klog.FromContext(ctx).WithValues("resourceName", key)
```

Comment:
This should get flagged by logcheck: instead of `WithValues`, use `klog.LoggerWithValues`.

Reply:
I ran logcheck and it didn't flag this at all. Fixed in recent commit.

Reply:
Did you run it locally? The "with-helpers" check must be enabled explicitly for this one, but pull-kubernetes-verify [should do that](https://github.com/kubernetes/kubernetes/blob/45279654db87f4908911569c07afc42804f0e246/hack/logcheck.conf#L31-L35).

Speaking of that job, there is one linter error:
```
staging/src/k8s.io/sample-controller/controller.go:252:2: ineffectual assignment to ctx (ineffassign)
	ctx = klog.NewContext(ctx, logger)
	^
```

Reply:
I was running it in stand-alone mode. When I ran it with verify-golangci-lint.sh it worked. Thanks for the tip!. Now it's clean.